rpi-eeprom-update

```diff
@@ -54,50 +61,85 @@ cleanup() {
       rm -f "${TMP_EEPROM_IMAGE}"
    fi
-   if [ -f "${TMP_EEPROM_CONFIG}" ]; then
-      rm -f "${TMP_EEPROM_CONFIG}"
+   if [ -f "${TMP_CURRENT_EEPROM_CONFIG}" ]; then
+      rm -f "${TMP_CURRENT_EEPROM_CONFIG}"
+   fi
+   if [ -f "${TMP_NEW_EEPROM_CONFIG}" ]; then
+      rm -f "${TMP_NEW_EEPROM_CONFIG}"
    fi
    if [ -d "${TMP_BOOTFS_MNT}" ]; then
       umount "${TMP_BOOTFS_MNT}"
       rmdir "${TMP_BOOTFS_MNT}"
    fi
    TMP_BOOTFS_MNT=
    TMP_EEPROM_IMAGE=
-   TMP_EEPROM_CONFIG=
+   TMP_CURRENT_EEPROM_CONFIG=
+   TMP_NEW_EEPROM_CONFIG=
 }
 trap cleanup EXIT
 
-die() {
-   echo "$@" >&2
-   exit ${EXIT_FAILED}
-}
-
 prepareImage()
 {
    [ -f "${BOOTLOADER_UPDATE_IMAGE}" ] || die "EEPROM image \'${BOOTLOADER_UPDATE_IMAGE}\' not found"
    TMP_EEPROM_IMAGE="$(mktemp)"
-   TMP_EEPROM_CONFIG="$(mktemp)"
+   TMP_CURRENT_EEPROM_CONFIG="$(mktemp)"
 
-   mkdir -p "${FIRMWARE_BACKUP_DIR}"
+   vcgencmd bootloader_config > "${TMP_CURRENT_EEPROM_CONFIG}"
 
    # Backup the configuration of the currently loaded bootloader
-   vcgencmd bootloader_config > "${TMP_EEPROM_CONFIG}"
+   mkdir -p "${FIRMWARE_BACKUP_DIR}"
    backup="${FIRMWARE_BACKUP_DIR}/pieeprom-backup-$(date +%Y%m%d-%H%M%S).conf"
-   cp -f "${TMP_EEPROM_CONFIG}" "${backup}"
-
-   if [ "$(wc -l "${TMP_EEPROM_CONFIG}" | awk '{print $1}')" -lt 3 ]; then
-      # Don't propagate empty EEPROM config files and also prevent the initial
-      # bootloader config with WAKE_ON_GPIO=0 propgating to newer versions by
-      # accident.
-      OVERWRITE_CONFIG=1
-   fi
+   cp -f "${TMP_CURRENT_EEPROM_CONFIG}" "${backup}"
 
    cp -f "${BOOTLOADER_UPDATE_IMAGE}" "${TMP_EEPROM_IMAGE}"
 
-   if [ "${OVERWRITE_CONFIG}" = 0 ]; then
+   if [ "${USE_EEPROM_INTERNAL_CONFIG}" = 0 ]; then
+      TMP_NEW_EEPROM_CONFIG="$(mktemp)"
+      rpi-eeprom-config "${TMP_EEPROM_IMAGE}" --out "${TMP_NEW_EEPROM_CONFIG}"
+
+      # Clean up empty lines
+      sed -i -e 's/[\t ]//g;/^$/d' "${TMP_NEW_EEPROM_CONFIG}"
+
+      no_of_eeprom_config_properties="$(wc -l "${TMP_CURRENT_EEPROM_CONFIG}" | awk '{print $1}')"
+      if [ "$no_of_eeprom_config_properties" -lt 3 ]; then
+         # Prevent the initial bootloader config with WAKE_ON_GPIO=0
+         # propagating to newer versions by accident
+         sed -i -e 's/WAKE_ON_GPIO=.*/WAKE_ON_GPIO=1/1' "${TMP_NEW_EEPROM_CONFIG}"
+      fi
+
+      # Apply current config settings to new EEPROM config
+      while read line; do
+         field=$(echo "${line}" | cut -d'=' -f1)
+         sed -i -e "s/${field}=.*/${line}/1" "${TMP_NEW_EEPROM_CONFIG}"
+      done < "${TMP_CURRENT_EEPROM_CONFIG}"
+
+      # Apply user customisations to new EEPROM config
+      if [ -f "${CUSTOM_EEPROM_CONFIG}" ]; then
+         echo "\n*** APPLYING CUSTOM EEPROM CONFIGURATION ***"
+         while read line; do
+            field="$(echo "${line}" | cut -d'=' -f1)"
+
+            if grep -q "${field}" "${TMP_NEW_EEPROM_CONFIG}"; then
+               if [ "${field}" = "FREEZE_VERSION" ]l; then
```

Comment:
Is that `l` just before the semi-colon a typo?

Reply:
Yes 😓